### PR TITLE
Keep previous default resources

### PR DESCRIPTION
**Pull request by danielrbradley**
The intention when moving to v2 is that all the default resources in v1 would be available as explicit versions.

Somewhere along the way most of these go squeezed out. This PR ensures that any explicit version from the previous versions's lock is included as an explicit version's resource.

This adds approximately 1776 resources to each SDK (as well as corresponding get methods).

Fixes https://github.com/pulumi/pulumi-azure-native/issues/2634

### CODE REVIEW

File: provider/pkg/versioning/gen.go [hunk 1]
```diff
@@ -37,6 +37,13 @@ func (v VersionMetadata) ShouldInclude(provider string, version string, typeName
 			}
 		}
 	}
+	if resources, ok := v.PreviousLock[provider]; ok {
```
Comment (thomas11) [inline provider/pkg/versioning/gen.go:40]: This repeated triple-nested lookup could be a nice helper on DefaultVersionLock

Comment (thomas11) [inline provider/pkg/versioning/gen.go:40]: I think this warrants a comment like "Include all resources in the previous major version's lock so users are not forced to upgrade"

**Review by thomas11 — APPROVED** [review]

**Comment by mikhailshilkov** [comment]
Sorry I didn't review it on time. 

This looks about right to me. Questions:

- Can we add tests to ShouldInclude to cover this change?
- Can we amend one of ProgramTest tests to cover a version of this change?
- What is the impact on SDK size?

**Comment by mikhailshilkov** [comment]
One more question: 

> schema-tools stats -p azure-native

says we have 

> "TotalResources": 1658

How is it possible this PR adds 1776 resources?

**Comment by danielrbradley** [comment]
Schema tools stats doesn't include explicit versions as it's only the docs schema.

It looks like it was actually closer to 1,200 resources based on the the SDK diff changes.

**Pull request merged**